### PR TITLE
The lite-core module decouples ZooKeeper

### LINKED ISSUE #2183: The lite-core module decouples ZooKeeper
## Feature Request

The lite-core module decouples ZooKeeper.

### PR DESCRIPTION
**Pull request by mytang0**
Fixes #2183 .

Changes proposed in this pull request:

The elasticjob-lite-core module decouples ZooKeeper, introduce the implementation of the on-demand registration center when using it.

### CODE REVIEW

**Comment by mytang0** [comment]
Please take a look @TeslaCN

File: elasticjob-lite/elasticjob-lite-core/src/main/java/org/apache/shardingsphere/elasticjob/lite/api/registry/JobInstanceRegistry.java [hunk 1]
```diff
@@ -17,9 +17,14 @@
 
 package org.apache.shardingsphere.elasticjob.lite.api.registry;
 
+import java.util.Arrays;
```
Comment (TeslaCN) [inline elasticjob-lite/elasticjob-lite-core/src/main/java/org/apache/shardingsphere/elasticjob/lite/api/registry/JobInstanceRegistry.java:20]: There are too many unrelated changes in this PR. Please revert them.

File: elasticjob-lite/elasticjob-lite-core/pom.xml [hunk 2]
```diff
@@ -125,5 +121,11 @@
             <artifactId>logback-classic</artifactId>
             <scope>test</scope>
         </dependency>
+        <dependency>
+            <groupId>org.apache.shardingsphere.elasticjob</groupId>
+            <artifactId>elasticjob-registry-center-zookeeper-curator</artifactId>
+            <version>${project.parent.version}</version>
+            <scope>test</scope>
```
Comment (TeslaCN) [inline elasticjob-lite/elasticjob-lite-core/pom.xml:128]: Could we consider setting it `runtime`? Otherwise users need to introduce it sperately.

**Review by TeslaCN — CHANGES_REQUESTED** [review]

File: elasticjob-lite/elasticjob-lite-core/pom.xml (hunk 2 above)
Comment (mytang0) [reply elasticjob-lite/elasticjob-lite-core/pom.xml:128]: Yes, after decoupling, users are expected to actively introduce.

File: elasticjob-lite/elasticjob-lite-core/src/main/java/org/apache/shardingsphere/elasticjob/lite/api/registry/JobInstanceRegistry.java (hunk 1 above)
Comment (mytang0) [reply elasticjob-lite/elasticjob-lite-core/src/main/java/org/apache/shardingsphere/elasticjob/lite/api/registry/JobInstanceRegistry.java:20]: Caused by reformatting code.

**Review by linghengqian — COMMENTED** [review]
- Will this PR be on the ElasticJob 3.0.4 milestone?

**Review by linghengqian — COMMENTED** [review]
- @mytang0 Do you still have time to update the current PR? Maybe allow another person to raise a new PR fixing the current one? I just noticed https://github.com/apache/shardingsphere-elasticjob/issues/2400 .
